pkg/chunk/storage/storage_client_test.go

```diff
@@ -66,3 +58,53 @@ func TestChunksBasic(t *testing.T) {
 		}
 	})
 }
+
+func createChunks(startIndex, batchSize int) ([]string, []chunk.Chunk, error) {
+	keys := []string{}
+	chunks := []chunk.Chunk{}
+	for j := 0; j < batchSize; j++ {
+		chunk := dummyChunkFor(model.Now(), model.Metric{
+			model.MetricNameLabel: "foo",
+			"index":               model.LabelValue(strconv.Itoa(startIndex*batchSize + j)),
+		})
+		chunks = append(chunks, chunk)
+		_, err := chunk.Encode() // Need to encode it, side effect calculates crc
+		if err != nil {
+			return nil, nil, err
+		}
+		keys = append(keys, chunk.ExternalKey())
+	}
+	return keys, chunks, nil
+}
+
+type clientWithErrorParameters interface {
+	SetErrorParameters(provisionedErr, errAfter int)
+}
+
+func TestChunksPartialError(t *testing.T) {
+	forAllFixtures(t, func(t *testing.T, client chunk.StorageClient) {
+		// This test is currently very specialised for DynamoDB
+		if !strings.Contains(t.Name(), "DynamoDB") {
```

Comment:
Perhaps this test belongs in `aws` package instead then?  We already put the dynamodb specific table manager tests there.

Reply:
I tried. It would need more mechanism to get to the fixture stuff, while avoiding circular dependencies.

Reply:
I had another go; after 2 hours it's still not passing tests so going to merge this one and post the refactor as a separate PR.